phoenicis-scripts/src/main/java/org/phoenicis/scripts/engine/ScriptEngineType.java

```diff
@@ -1,17 +1,34 @@
 package org.phoenicis.scripts.engine;
 
+import javax.script.ScriptEngineManager;
+import java.util.Map;
+
 /**
  * type of the script engine
  */
 public enum ScriptEngineType {
-    NASHORN("nashorn"), GRAAL("graal.js");
+    NASHORN("nashorn") {
+        @Override
+        public PhoenicisScriptEngine createScriptEngine() {
+            return new JSAScriptEngine(new ScriptEngineManager().getEngineByName("nashorn"));
+        }
+    },
+
+    GRAAL("graal.js") {
+        @Override
+        public PhoenicisScriptEngine createScriptEngine() {
+            return new PolyglotScriptEngine("js", Map.of("js.nashorn-compat", "true"));
+        }
+    };
 
     private final String name;
 
     ScriptEngineType(String name) {
         this.name = name;
     }
 
+    public abstract PhoenicisScriptEngine createScriptEngine();
```

Comment:
Do we want this in an enum? I'd rather put this functionality in the factory.

Reply:
I also thought about putting this in the factory, my issue was where?
The problem was that `createEngine` in `PhoenicisScriptEngineFactory` does so much more than simply creating an engine object. It also initializes it, i.e. it adds the injectors.

After adding the method to the `ScriptEngineType` I also discovered that it was quite a nice approach, because you can specify the required parameters in the `ScriptEngineType` objects.
For example we could add a graal python engine easily by adding:

```java
    PYTHON("graal.python") {
        @Override
        public PhoenicisScriptEngine createScriptEngine() {
            return new PolyglotScriptEngine("python", Map.of());
        }
    };
```

This approach doesn't require extra `switch case` constructs or anything similar.

Reply:
It's just that you wouldn't expect functionality in an enum. Also, you basically moved the factory to the enum. I would really prefer `switch case` in the factory. Adding injectors in the factory is perfectly fine because its part of building the object (logically - that it doesn't happen in the constructor is only implementation details).

Reply:
I'm fine with moving this functionality to a different place. I just don't know where and how to do this in an elegant manner.

I would suggest to merge this as is for now and change it later on. 
I mean we still need to change the `PhoenicisScriptEngine` implementations multiple times. So we have enough time to think of a better place to instantiate them.

Reply:
Ok.